File: subprojects/frida-python/examples/open_service/dtx/screenshot.py
import sys

import frida

if len(sys.argv) != 2:
    print(f"Usage: {sys.argv[0]} outfile.png", file=sys.stderr)
    sys.exit(1)
outfile = sys.argv[1]

device = frida.get_usb_device()

screenshot = device.open_service("dtx:com.apple.instruments.server.services.screenshot")
png = screenshot.request({"method": "takeScreenshot"})
with open(outfile, "wb") as f:
    f.write(png)

File: subprojects/frida-node/examples/open_service/dtx/screenshot.ts
import frida from "frida";
import fs from "fs/promises";

async function main() {
    if (process.argv.length !== 3) {
        console.error(`Usage: ${process.argv[0]} outfile.png`);
        process.exit(1);
    }
    const outfile = process.argv[2];

    const device = await frida.getUsbDevice();

    const screenshot = await device.openService("dtx:com.apple.instruments.server.services.screenshot");
    const png = await screenshot.request({ method: "takeScreenshot" });
    await fs.writeFile(outfile, png);
}

main()
    .catch(e => {
        console.error(e);
    });
